calculator_test.go

```diff
@@ -2,23 +2,293 @@ package calculator_test
+			arg2: 10,
+			nums: []float64{10, 4},
+			want: 2.5,
+		},
+		{
+			name: "Incorrect division but no err expected",
+			arg1: 1000,
+			arg2: 10,
+			nums: []float64{10, 4},
+			want: 2.5,
+		},
+		{
+			name: "Irrational result is 'close enough'",
+			arg1: 1,
+			arg2: 3,
+			want: .33333,
+		},
+	}
+
+	for _, tc := range testCases {
+		got, err := calculator.Divide(tc.arg1, tc.arg2, tc.nums...)
+
+		errReceived := err != nil
+		if errReceived != tc.errExpected {
+			t.Fatalf("Divide(%f,%f,%v): unexpected error status %v in TestCase: %s", tc.arg1, tc.arg2, tc.nums, err, tc.name)
+		}
+		expectedResult := closeEnough(tc.want, got, 0.0001)
+		if !tc.errExpected && !expectedResult {
+			t.Errorf("Divide(%f,%f,%v): wanted %f, got %f in TestCase: %s", tc.arg1, tc.arg2, tc.nums, tc.want, got, tc.name)
+		}
+
+	}
+}
+
+func TestSqrt(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	var testCases = []struct {
+		name        string
+		a           float64
+		want        float64
+		errExpected bool
+	}{
+		{name: "Sqrt of negative results in an error", a: -1, want: 0, errExpected: true},
+		{name: "Sqrt of zero results in a zero", a: 0, want: 0},
+		{name: "Sqrt of positive number results in positive number", a: 16, want: 4},
+	}
+
+	for _, tc := range testCases {
+		got, err := calculator.Sqrt(tc.a)
+
+		errReceived := err != nil
+		if errReceived != tc.errExpected {
+			t.Fatalf("Sqrt(%f): unexpected error status %v in TestCase: %s", tc.a, err, tc.name)
+		}
+		expectedResult := closeEnough(tc.want, got, 0.0001)
+		if !tc.errExpected && !expectedResult {
+			t.Errorf("Sqrt(%f): wanted %f, got %f in TestCase: %s", tc.a, tc.want, got, tc.name)
+		}
+
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+func TestExpressionProcessor(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name        string
+		expr        string
+		want        float64
+		errExpected bool
+	}{
+		{
+			name: "Expression with no whitespace",
+			expr: "5*2",
+			want: 10,
+		},
+		{
+			name: "Expression with standard whitespace",
+			expr: "5 + 2",
+			want: 7,
+		},
+		{
+			name: "Expression with extra whitespace",
+			expr: "7  -   2",
+			want: 5,
+		},
+		{
+			name: "Division",
+			expr: "9 / 3",
+			want: 3,
+		},
+		{
+			name:        "Incomplete Expression",
+			expr:        "5*",
+			errExpected: true,
+		},
+		{
+			name:        "Malformed Expression (postfix)",
+			expr:        "5 7 *",
+			want:        35,
+			errExpected: true,
+		},
+	}
+	for _, tc := range testCases {
+		got, err := calculator.ExpressionProcessor(tc.expr)
+
+		errReceived := err != nil
+
+		if errReceived != tc.errExpected {
+			t.Fatalf("ExpressionProcessor(%s): unexpected error status %v in TestCase: %s", tc.expr, err, tc.name)
```

Comment:
Remember the test itself will print out the name of the test (`TestExpressionProcessor`), so we don't need to repeat that here, just the test case input.